### PR TITLE
ci: lava: remove hardcoded strings

### PR DESCRIPTION
**Pull request by mwasilew**
LAVA backend is able to automatically resubmit test jobs in case it
detects specific strings in LAVA error messages. These strings are now
part of settings instead of being hardcoded in the backend code.

Signed-off-by: Milosz Wasilewski <milosz.wasilewski@linaro.org>

### CODE REVIEW

**Comment by mwasilew** [comment]
I added second commit that moves settings to DB. Not sure if the first commit makes any sense in this situation.

**Comment by terceiro** [comment]
yeah, I would get rid of the first commit.

I would also use YAML instead of JSON as the former is less annoying to write.

File: squad/ci/backend/lava.py [hunk 1]
```diff
@@ -268,6 +266,9 @@ def __get_publisher_event_socket__(self):
         return self.proxy.scheduler.get_publisher_event_socket()
 
     def __parse_results__(self, data, test_job):
+        LAVA_INFRA_ERROR_MESSAGES = []
+        if self.settings is not None:
+            LAVA_INFRA_ERROR_MESSAGES = self.settings.get('CI_LAVA_INFRA_ERROR_MESSAGES', [])
```
Comment (terceiro) [inline squad/ci/backend/lava.py]: drop the usage of all caps?

File: squad/ci/backend/null.py [hunk 2]
```diff
@@ -1,3 +1,4 @@
+import json
```
Comment (terceiro) [inline squad/ci/backend/null.py]: IMO YAML is easier to writer

**Pull request merged**